### DISCUSSION TITLE
How do I create indexes when I add new vectors?

### DISCUSSION THREAD

**Discussion by skye95git** [milvus-io/milvus#16336] [Q&A and General discussion]
I've created a collection and inserted some vectors. Now I want to insert some new vectors, how do I create an index to avoid duplicate indexes? Do I call `create_index()` every time I insert new data?

**Comment by yhmo** [comment]
Milvus split data into segments, each segment size is defined by a configuration. In v1.x, segment size is defined by "index_file_size", it is a parameter of create_collection(). In v2.0, segment size is defined by milvus.yaml. After a segment size reach the threshold, milvus will build index for it.
So, after you only call create_index() one time, specify an index type to a collection, then milvus will automatically build index for segments that reach the threshold. If you continue to insert more data, mivus will accumulate the data into new segment, and the segment growing to reach the threshold, then milvus automatically build index for it.
When you call search(), if some segments have have no index, milvus will do brute-force search for them.
